src/ucs/sys/sock.c

```diff
@@ -383,10 +401,13 @@ ucs_socket_do_iov_nb(int fd, struct iovec *iov, size_t iov_cnt, size_t *length_p
     };
     ssize_t ret;
 
-    ucs_assert(iov_cnt > 0);
-
     ret = iov_func(fd, &msg, MSG_NOSIGNAL);
-    if (ucs_likely(ret > 0)) {
+    if (ucs_likely(/* The IO operation is considered as successful if: */
+                   /* - the return value > 0 */
+                   (ret > 0) ||
+                   /* - the number of user's IOV elements == 0 (Datagram
+                    *   sockets permit zero-length datagrams) */
+                   (iov_cnt == 0))) {
```

Comment:
need to check ret here as well, sorry did not notice at first review

Reply:
sorry, I forgot about this. fixed